src/vehicle.h

```diff
@@ -221,6 +226,9 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
     item base;
     std::list<item> items; // inventory
 
+    /** read only from legacy savegames */
+    int legacy_hp = -1;
```

Comment:
> It's private tovehicle_part

Oops, sorry, I misread the collapsed diff.

> and arguably the simplest implementation

It can be made even simpler: why do you need to _store_ it. You read the `legacy_hp` literally a couple of lines below the `base` item and presumably can do everything there.